discovery.go

```diff
@@ -0,0 +1,127 @@
+package viamrtsp
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/logging"
+)
+
+// RTSPDiscovery is responsible for discovering RTSP camera devices using WS-Discovery and ONVIF.
+type RTSPDiscovery struct {
+	multicastAddress string
+	logger           logging.Logger
+}
+
+// NewRTSPDiscovery creates a new RTSPDiscovery instance with default values.
+func NewRTSPDiscovery(logger logging.Logger) *RTSPDiscovery {
+	return &RTSPDiscovery{
+		multicastAddress: "239.255.255.250:3702", // Standard WS-Discovery multicast address
```

Comment:
I think this is the standard multicast address/port used for ONVIF devices, so we could probably get away with not having this configurable.

Reply:
Okay cool that's what I read on Wikipedia